alg/gdalgeoloc.cpp

```diff
@@ -231,13 +231,17 @@ static bool GeoLocGenerateBackMap( GDALGeoLocTransformInfo *psTransform )
     const size_t nXSize = psTransform->nGeoLocXSize;
     const size_t nYSize = psTransform->nGeoLocYSize;
 
-/* -------------------------------------------------------------------- */
-/*      Decide on resolution for backmap.  We aim for slightly          */
-/*      higher resolution than the source but we can't easily           */
-/*      establish how much dead space there is in the backmap, so it    */
-/*      is approximate.                                                 */
-/* -------------------------------------------------------------------- */
-    const double dfTargetPixels = (static_cast<double>(nXSize) * nYSize * OVERSAMPLE_FACTOR);
+/* ------------------------------------------------------------------- */
+/*      Decide on resolution for backmap.  We make sure that the       */
+/*      X and Y resolutions are at least OVERSAMPLE_FACTOR higher than */
+/*      the source.                                                    */
+/* ------------------------------------------------------------------- */
+    // Higher bound for pixel size
+    const double nPixelSizeBound = std::min((psTransform->dfMaxX - psTransform->dfMinX) / nXSize,
+                                            (psTransform->dfMaxY - psTransform->dfMinY) / nYSize) / OVERSAMPLE_FACTOR;
+    const double dfTargetPixels =
+      (psTransform->dfMaxX - psTransform->dfMinX) *
+      (psTransform->dfMaxY - psTransform->dfMinY) / (nPixelSizeBound * nPixelSizeBound);
     const double dfPixelSize = sqrt(
```

Comment:
My test case source resolution is (4096, 1000) and the backmap resolution computed by the original algorithm was (4843,846) - the number of pixels in the backmap increased but the resolution along Y axis decreased. The proposed change make sure that this does not happen - but potentially increases the backmap size.